grub-core/loader/i386/linux.c

```diff
@@ -1152,19 +1152,80 @@ grub_cmd_initrd (grub_command_t cmd __attribute__ ((unused)),
   return grub_errno;
 }
 
+static grub_err_t
+grub_cmd_measure (grub_command_t cmd __attribute__ ((unused)),
+		int argc, char *argv[])
+{
+  grub_file_t file = 0;
+  grub_ssize_t len;
+  grub_uint8_t index;
+  grub_uint8_t *read_data = NULL;
+
+  if (argc != 2)
+    {
+      grub_error (GRUB_ERR_BAD_ARGUMENT, N_("filename and PCR expected"));
+      goto fail;
+    }
+
+  index = grub_strtoul(argv[1], NULL, 10 );
+
+  /* if index is invalid */
+  if( grub_errno != GRUB_ERR_NONE )
+    {
+      grub_error (GRUB_ERR_BAD_ARGUMENT, N_("filename and PCR expected"));
+      goto fail;
+    }
+
+  file = grub_file_open (argv[0]);
+  if (! file)
+    goto fail;
+
+  len = grub_file_size (file);
+  read_data = grub_malloc (len);
+
+  if (!read_data)
+    {
+      grub_error (GRUB_ERR_OUT_OF_MEMORY, N_("cannot allocate data buffer"));
+      goto fail;
+    }
+
+  if (grub_file_read (file, read_data, len) != len)
+    {
+      if (!grub_errno)
```

Comment:
definitely use braces, and make sure to keep the same indentation style (even though GNU style is the worst style)